lib/plan-builder-base.js

```diff
@@ -59,7 +59,7 @@ function castArg(arg, funcName, paramName, argPos, paramTypes) {
       arg = arg.valueOf();
     } else if (arg instanceof types.ServerType) {
       if(arg._ns === 'vec'){
-        return arg._args;
+        return arg;
```

Comment:
Whatever the right line of code is here, we definitely need a comment explaining why it's okay not to throw an error if the namespace is "vec", and then why the arg should actually be the value of "_args" (the previous code) or why it doesn't need to be cast at all. This "if ServerType" block is particularly confusing for a new reader (me) because there's no explanation too as to why an Error was being thrown before, which presumably would only have been in the case of programmer error.